napari/layers/surface/surface.py

```diff
@@ -286,10 +293,29 @@ def _set_view_slice(self):
         disp = list(self.dims.displayed)
         indices = np.array(self.dims.indices)
 
+        # Take vertex_values dimensionality into account.
+        # Note that displayed dimensions corresponding to vertex_values will be
+        # ignored from the display and averaged over to get the used values
```

Comment:
I don't understand this comment, nor the code underneath it... Would you mind adding more comments to explain further what this is doing? I imagine "adj" means "adjusted", but adjusted for what? What averaging is going on?

Reply:
I've added more comments, hope it helps, if not I can add more. Agreed it's a little confusing here